### DISCUSSION TITLE
Serializing/Deserializing multiple data structures into a single CSV file

### DISCUSSION THREAD

**Discussion by ThomasAlban** [BurntSushi/rust-csv#369] [Q&A]
I want to be able to serialize 2 data structures: The first is a `Transform` struct, and the second is a generic struct `T: Serialize + Deserialize`. Depending on some settings I only want to serialize certain parts of the `Transform` struct, so e.g. some may not serialize the scale/rotation. These settings are determined by the type `T`.

So, when serializing these structs:
```rust
Transform {
    position: Vec3,
    rotation: Vec3,
   scale: Vec3,
}
GenericStructPassedIn {
    first_field: f32,
    second_field: f32,
}
SettingsForThatGenericStruct {
    do_rotation: bool,
    do_scale: bool,
}
```
I want to end up with a csv that looks like this:
```
position_x,position_y,position_z,rotation_x,rotation_y,rotation_z,first_field,second_field
0.0,1.0,2.0,0.0,0.0,0.0,4.5,3.9,
...etc
```
And similarly, I want to be able to deserialize that csv and go back to the structs above. I've done some research but can't seem to find a way of doing this.
Any help appreciated!

**Comment by BurntSushi** [comment]
You could try defining a parent struct with each of your structs as fields and apply `serde(flatten)` to them. But the issue tracker is full of problems with `serde(flatten)`, so I'm not sure how well that will work.

Otherwise, I would define a separate "wire" type that matches your CSV record format and use `serde(try_from)` and `serde(into)` to have serde do conversions for you automatically.
